src/shared/components/palette/case-flag/components/select-flag-type/select-flag-type.component.ts

```diff
@@ -0,0 +1,81 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { CaseFlagFieldState, SelectFlagTypeErrorMessage } from '../../enums';
+import { ErrorMessage } from '../../../../../domain';
+import { CaseFlagState, FlagType } from '../../domain';
+
+@Component({
+  selector: 'ccd-select-flag-type',
+  templateUrl: './select-flag-type.component.html',
+  styleUrls: ['./select-flag-type.component.scss']
+})
+export class SelectFlagTypeComponent implements OnInit {
+
+  @Input()
+  public formGroup: FormGroup;
+
+  @Output()
+  public caseFlagStateEmitter: EventEmitter<CaseFlagState> = new EventEmitter<CaseFlagState>();
+
+  public flagTypes: FlagType[];
+  public flagTypeSelected: string;
+  public errorMessages: ErrorMessage[];
+  public flagTypeNotSelectedErrorMessage = '';
+  public flagTypeErrorMessage = '';
+
+  public ngOnInit(): void {
+    this.flagTypes = this.getFlagTypes();
+    this.flagTypes.forEach(flagType => {
+      this.formGroup.addControl(flagType.id, new FormControl(''));
+    });
+    this.formGroup.addControl('other', new FormControl(''));
+    this.formGroup.addControl('otherFlagTypeDescription', new FormControl(''));
+  }
+
+  public onFlagTypeChanged(flagTypeId: string): void {
+    // Display description textbox if 'other' checkbox is selected
+    this.flagTypeSelected = flagTypeId;
+  }
+
+  public onNext(): void {
+    // Validate form
+    this.validateForm();
+    // Return case flag field state and error messages to the parent
+    this.caseFlagStateEmitter.emit({ currentCaseFlagFieldState: CaseFlagFieldState.FLAG_TYPE, errorMessages: this.errorMessages });
+  }
+
+  private validateForm(): boolean {
+    this.flagTypeNotSelectedErrorMessage = '';
+    this.flagTypeErrorMessage = '';
+    this.errorMessages = [];
+
+    if (!this.flagTypeSelected) {
+      this.flagTypeNotSelectedErrorMessage = SelectFlagTypeErrorMessage.FLAG_TYPE_NOT_SELECTED;
+      this.errorMessages.push({title: '', description: `${SelectFlagTypeErrorMessage.FLAG_TYPE_NOT_SELECTED}`, fieldId: 'conditional-radios-list'})
+      return false;
+    }
+    if (this.flagTypeSelected === 'other') {
+      const otherFlagTypeDescription = this.formGroup.get('otherFlagTypeDescription').value;
+      if (!otherFlagTypeDescription) {
+        this.flagTypeErrorMessage = SelectFlagTypeErrorMessage.FLAG_TYPE_NOT_ENTERED;
+        this.errorMessages.push({title: '', description: `${SelectFlagTypeErrorMessage.FLAG_TYPE_NOT_ENTERED}`, fieldId: 'other-flag-type-description'});
+        return false;
+      }
+      if (otherFlagTypeDescription.length > 80) {
```

Comment:
@johnbenjamin-hmcts - you may want to move the `80` into its own private static variable so that it doesn't look like a "magic number".

Reply:
@Daniel-Lam - I have made the changes as you suggested.